functions/v2/helloworld/hello_pubsub_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_cloudevent_pubsub_unit_test]
+
+package helloworld
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+)
+
+func TestHelloPubSub(t *testing.T) {
```

Comment:
This looks good, but here is also a Go function framework. Should we be using that at all. I am not sure if it has been updated for working in this new environment.(If it needs updates)

Reply:
I'll have to let @grayside  weigh in on this.  The other language samples didn't use the function framework.

Reply:
If they did not either, I think this is fine. It is probably best to be consistent between the samples.

Reply:
In my opinion, we should not use the function framework in unit tests unless it would save us from more complex mocking.

Using the framework explicitly in tests would be something like a high overhead, local system test. I think it should be a lower priority than a deploy-based test, only created for specific need.

Deploy tests are waiting on our team sorting out a better testing strategy.

Reply:
ack.